otelgoyek/example_test.go

```diff
@@ -0,0 +1,262 @@
+package otelgoyek_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+
+	"github.com/goyek/goyek/v2"
+	"github.com/goyek/goyek/v2/middleware"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/sdk/trace"
+
+	"github.com/goyek/x/otelgoyek"
+)
+
+func run(ctx context.Context, w io.Writer, tasks []string) (err error) {
+	// Handle SIGINT (CTRL+C) gracefully.
+	ctx, stop := signal.NotifyContext(ctx, os.Interrupt)
+	go func() {
+		<-ctx.Done()
+		fmt.Fprintln(w, "first interrupt, graceful stop")
+		stop()
+	}()
+
+	// Setup OpenTelemetry tracing pipeline.
+	traceExporter, err := stdouttrace.New(stdouttrace.WithPrettyPrint())
+	if err != nil {
+		return err
+	}
+	tracerProvider := trace.NewTracerProvider(trace.WithBatcher(traceExporter))
+	defer func() {
+		err = errors.Join(err, tracerProvider.Shutdown(context.Background()))
+	}()
+	otel.SetTracerProvider(tracerProvider)
```

Comment:
I wonder if it wouldn't be better to show in the example how to use `WithTracerProvider` option instead of relying on otel's global one. Wdut?

Reply:
I was thinking about it as well. The main think is that using global providers is a common practice in OTel and also `goyek` uses globals for convivence. While personally I dislike using global variables, I see that using them can make the usage easier.